docs: GitHub integration (#1869)
neondatabase/website: content/docs/guides/neon-github-integration.md

---
title: The Neon GitHub integration
subtitle: Connect Neon Postgres to a GitHub repository and build GitHub Actions workflows
enableTableOfContents: true
redirectFrom:
  - /docs/guides/neon-github-app
updatedOn: '2024-06-14T07:55:54.401Z'
---

<Admonition type="comingSoon" title="Feature Coming Soon">
The Neon GitHub integration is currently in **private preview**. To start using it, request access by contacting our [Customer Success](mailto:customer-success@neon.tech) team and asking to join the private preview.
</Admonition>

> The Neon GitHub integration installs a GitHub App that connects your Neon project to a GitHub repository by setting a Neon API key secret and Neon project ID variable in your GitHub repository. The integration also provides a sample GitHub Actions workflow that uses the configured secret and variable to create a database branch for each pull request.

Reviewer: ```suggestion
The Neon GitHub integration connects your Neon projects to corresponding GitHub repositories, helping you incorporate your database into your developer workflow. For example, create a database branch for each pull request using GitHub Actions. We’ll provide you with a sample GitHub Actions workflow to help you get started with that.

## How it works

The integration installs the GitHub App, letting you select which repositories you want to make accessible to Neon. When you connect a particular project to an available repository in GitHub, the integration uses the GitHub App to set a Neon API key secret and a Neon project ID variable in the selected repository. Your project is now connected, and you can use GitHub Actions to include this project’s database in your workflow.
```

   <Admonition type="note">
   For GitHub to discover GitHub Actions workflows, you must save the workflow files in a directory called `.github/workflows` in your repository. You can name the workflow file whatever you like, but you must use `.yml` or `.yaml` as the file name extension.
   </Admonition>

2. Copy the workflow code into your `neon-workflow.yml` file.
3. Commit your changes.

### Using the GitHub Actions workflow

To use the sample workflow, create a pull request in your GitHub application repository. This will trigger the `Create Neon Branch` action. You can verify that a branch was created on the **Branches** page in the Neon Console. You should see a new branch with a `preview/pr-` name prefix.

Closing the pull request removes the Neon branch from the Neon project, which you can also verify on the **Branches** page in the Neon Console.

To view workflow results in GitHub, follow the instructions in [Viewing your workflow results](https://docs.github.com/en/actions/quickstart#viewing-your-workflow-results), in the _GitHub documentation_.

## Building your own GitHub Actions workflow

> The sample workflow provided by the GitHub integration serves as a template, which you can expand upon or customize. The workflow uses Neon's create and delete branch GitHub Actions, which you can find here:

Reviewer: ```suggestion
The sample workflow provided by the GitHub integration serves as a template, which you can expand on or customize. The workflow uses Neon's create and delete branch GitHub Actions, which you can find here:
```

3. Select the repository that you chose when installing the Neon GitHub integration.
4. On the repository page, select the **Settings** tab.
5. Select **Secrets and variables** > **Actions** from the sidebar.

Your `NEON_API_KEY` secret is listed on the **Secrets** tab, and the `NEON_PROJECT_ID` variable is listed on the **Variables** tab.

## Disconnect a Neon project from a GitHub repository

Disconnecting a Neon project from a GitHub repository performs the following actions for the Neon project:

- Removes the Neon API key created for this integration from your Neon account.
- Removes the GitHub secret containing the Neon API key from the associated GitHub repository.
- Removes the GitHub variable containing your Neon project ID from the associated GitHub repository.

Any GitHub Actions workflows you've added to the GitHub repository that are dependent on these secrets and variables will no longer work.

To disconnect your Neon project:

1. In the Neon Console, navigate to the **Integrations** page for your project.
2. Locate the GitHub integration and click **Manage** to open the **GitHub integration** drawer.
3. Click **Disconnect**.

## Remove the GitHub integration

> Removing the GitHub performs the following actions for all Neon projects that you connected to a GitHub repository using the GitHub integration:

Reviewer: ```suggestion
Removing the GitHub Integration performs the following actions for all Neon projects that you connected to a GitHub repository using the GitHub integration:
```
